src/overlays/ovl_i4/fox_sz.c

```diff
@@ -120,132 +129,142 @@ void SectorZ_Missile_Update(Actor* this) {
         gPlayer[0].csState = 100;
-        gActors->state = -31072;
+        gActors[0].state = -31072;
     }
+    // clang-format on
 }
 
-void SectorZ_80199FCC(Actor* actor, s32 arg1) {
-    Actor_Initialize(actor);
-    actor->obj.status = OBJ_INIT;
-    actor->obj.id = OBJ_ACTOR_ALLRANGE;
-    actor->aiType = AI360_MISSILE;
+void SectorZ_SpawnMissile(ActorAllRange* this, s32 missileWaveIdx) {
+    Actor_Initialize(this);
+    this->obj.status = OBJ_INIT;
+    this->obj.id = OBJ_ACTOR_ALLRANGE;
+    this->aiType = AI360_MISSILE;
 
-    actor->obj.pos.x = D_i4_8019F4C0[arg1].x;
-    actor->obj.pos.y = D_i4_8019F4C0[arg1].y;
-    actor->obj.pos.z = D_i4_8019F4C0[arg1].z;
+    this->obj.pos.x = sMissileWaveInitPos[missileWaveIdx].x;
+    this->obj.pos.y = sMissileWaveInitPos[missileWaveIdx].y;
+    this->obj.pos.z = sMissileWaveInitPos[missileWaveIdx].z;
 
-    actor->state = 5;
-    actor->rot_0F4.y = 180.0f;
+    this->state = 5;
+    this->rot_0F4.y = 180.0f;
 
-    Object_SetInfo(&actor->info, actor->obj.id);
+    Object_SetInfo(&this->info, this->obj.id);
 
-    actor->health = 250;
-    actor->info.drawType = 2;
-    actor->info.hitbox = SEGMENTED_TO_VIRTUAL(D_SZ_6009230);
-    actor->fwork[1] = 25.0f;
-    actor->fwork[29] = 2.0f;
+    this->health = 250;
+    this->info.drawType = 2;
+    this->info.hitbox = SEGMENTED_TO_VIRTUAL(aSZMissileHitbox);
+    this->fwork[1] = 25.0f;
+    this->fwork[29] = 2.0f;
 
-    AUDIO_PLAY_SFX(NA_SE_EN_PUNCH_ENGINE, actor->sfxSource, 4);
+    AUDIO_PLAY_SFX(NA_SE_EN_PUNCH_ENGINE, this->sfxSource, 4);
 }
 
-void SectorZ_8019A0F8(Actor* actor, s32 arg1) {
-    Actor_Initialize(actor);
-    actor->obj.status = OBJ_INIT;
-    actor->obj.id = OBJ_ACTOR_ALLRANGE;
-    actor->aiType = arg1 + AI360_10 + 3;
+// Enemies that follow the missile
+void SectorZ_MissileEscort(ActorAllRange* this, s32 enemyIndex) {
+    Actor_Initialize(this);
+    this->obj.status = OBJ_INIT;
+    this->obj.id = OBJ_ACTOR_ALLRANGE;
+    this->aiType = enemyIndex + AI360_10 + 3;
 
-    actor->obj.pos.x = gActors[10].obj.pos.x + D_i4_8019F4E4[arg1].x;
-    actor->obj.pos.y = gActors[10].obj.pos.y + D_i4_8019F4E4[arg1].y;
-    actor->obj.pos.z = gActors[10].obj.pos.z + D_i4_8019F4E4[arg1].z;
+    this->obj.pos.x = gActors[10].obj.pos.x + sMissileEscortOffsetPos[enemyIndex].x;
+    this->obj.pos.y = gActors[10].obj.pos.y + sMissileEscortOffsetPos[enemyIndex].y;
+    this->obj.pos.z = gActors[10].obj.pos.z + sMissileEscortOffsetPos[enemyIndex].z;
 
-    actor->state = 5;
-    actor->rot_0F4.y = 180.0f;
+    this->state = 5;
+    this->rot_0F4.y = 180.0f;
 
-    Object_SetInfo(&actor->info, actor->obj.id);
+    Object_SetInfo(&this->info, this->obj.id);
 
-    actor->fwork[1] = gActors[10].fwork[1];
-    actor->iwork[11] = 1;
+    this->fwork[1] = gActors[10].fwork[1];
+    this->iwork[11] = 1;
 }
 
-void SectorZ_8019A1D0(void) {
-    Actor* actor = &gActors[8];
+void SectorZ_Katt_Init(void) {
```

Comment:
at some point we should come up with a consistent scheme for Init, Setup, Spawn, etc.